### DISCUSSION TITLE
unable to write in the cowrie file system in the docker conatiner

### DISCUSSION THREAD

**Discussion by 0hex7** [telekom-security/tpotce#1426] [Q&A]
I have installed tpot as a standalone and i wanted to configure the honeypot to run only the cowrie honeypot.
i changed the tpot.yml file and ensure that only cowrie is running, now when i try to change the fs in cowrie, which is easily distinguishable as a honeypot to a new file system by using pickle file and a few other files, i am getting an error which says the file system is only readable.

e6f9629f10c3:~/cowrie$ ./bin/createfs -l YOUR-DIR -d 6 -o ./share/cowrie/custom.pickle
Traceback (most recent call last):
  File "/home/cowrie/cowrie/./bin/createfs", line 192, in <module>
    pickle.dump(tree, open(output, "wb"))
OSError: [Errno 30] Read-only file system: './share/cowrie/custom.pickle'
e6f9629f10c3:~/cowrie$ 

whenever i try to edit or configure something, even if i want to make a directory using mkdir, i get the error saying..
OSError: [Errno 30] Read-only file system: './share/cowrie/custom.pickle'

i tried to change the file in the docker-compose file also,
![image](https://github.com/telekom-security/tpotce/assets/108691415/b21eb504-01e5-4144-8918-2f8fbec5b560)
in the screenshot you can see that the read only file is set to true, i changed it to false and restarted the tpot, and even then i am getting the same error.

I am trying to implement a few lures in the fs using pickle fs, so if any of you can help me how to make the file system in conatiner writable, i will be eternally grateful

**Comment by 0hex7** [comment]
e6f9629f10c3:~/cowrie$ mkdir cat

mkdir: can't create directory 'cat': Read-only file system

e6f9629f10c3:~/cowrie$

**Comment by 0hex7** [comment]
any help would be great guys, i'm struct at this for weeks together

**Comment by 0hex7** [comment]
/home/cowrie/cowrie/share/cowrie/fs.pickle

Kippo/Cowrie file system interactive editor
Donovan Hubbard, Douglas Hubbard, March 2013
Type 'help' for help

fs.pickle:/$ ls
bin/
boot/
dev/
etc/
home/
initrd.img
lib/
lost+found/
media/
mnt/
opt/
proc/
root/
run/
sbin/
selinux/
srv/
sys/
test2
tmp/
usr/
var/
vmlinuz
fs.pickle:/$ who
*** Unknown syntax: who
fs.pickle:/$ nano
*** Unknown syntax: nano
fs.pickle:/$ mkdir cat
Unable to save pickle file '/home/cowrie/cowrie/share/cowrie/fs.pickle'. Are you sure you have write access?
[Errno 30] Read-only file system: '/home/cowrie/cowrie/share/cowrie/fs.pickle'
043e4f95c90d:~/cowrie/bin$

**Comment by 0hex7** [comment]
@t3chn0m4g3 Or @everyone please let me know if any solutions exist for the issue, its for my project and i need to submit it soon, i do not have much time guys ; (

**Reply by M0ki1** [reply]
This is honestly very late but, i changed the read_only to false and then build again the image, that fix the error, it was the same one you had, hope it helps
